modules/va_forms/app/controllers/va_forms/v0/forms_controller.rb

```diff
@@ -6,7 +6,44 @@ class FormsController < ApplicationController
       skip_before_action(:authenticate)
 
       def index
-        render json: Form.search(search_term: params[:query]),
+        if Flipper.enabled?(:new_va_forms_search)
+          if params[:query].present?
+            # The regex below checks to see if a form follows the DD-DDDD format (with optional alpha characters)
+            va_prefix_regex = /^\d{2}(?:[pP])?[- \s]\d+(?:[a-zA-Z])?$/
+            sf_form_regex = /^[sS][fF](?:[- \s])?\d+(?:[a-zA-Z])?$/
+            return search_by_form_number if params[:query].match(va_prefix_regex)
+                                                          .present? || params[:query]
+                                            .match(sf_form_regex)
+                                            .present?
```

Comment:
This inline `if` statement is kinda hard to read because the conditionals are kinda long and broken up on multiple lines. I think it might be better to make it regular `if` statement for readability

```ruby
if params[:query].match(va_prefix_regex).present? || params[:query].match(sf_form_regex).present?
  return search_by_form_number
end
```

Reply:
Done